map/src/main/java/openlr/map/utils/BearingPointCalculator.java

```diff
@@ -0,0 +1,106 @@
+     * Verify the relevant end (Start/End node) of the line is not an intersection
+     *
+     * @param connectedLines lines connected to relevant end
+     * @return true: if there is only one line connected to the node
+     * false: if there are more line connected to the node
+     */
+    private boolean isNotIntersection(int connectedLines) {
+        return (connectedLines == 1);
+    }
+
+
+    /**
+     * Lines connected to the relevant end of the line
+     *
+     * @param line                parent line
+     * @param inLocationDirection true: if bearing direction is same as the overall location direction
+     * @return if 'inLocationDirection == true' outgoing lines of the end node
+     * else if 'inLocationDirection == false' incoming lines of the start node
+     */
+    private List<Line> calculateNextLinesInRoute(Line line, boolean inLocationDirection) {
+        Iterator<Line> successors;
+        if (inLocationDirection) {
+            successors = line.getNextLines();
+        } else {
+            successors = line.getPrevLines();
+        }
+        return StreamSupport
+                .stream(Spliterators.spliteratorUnknownSize(successors, 0), false)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * @param line                line on which bearing destination exist
+     * @param offset              distance from the relevant end, based on the location direction, of the line to the bearing destination
+     * @param inLocationDirection true: if bearing direction is same as the overall location direction
+     * @return bearing destination coordinates
+     */
+    private GeoCoordinates calculateBearingDestinationOnLine(Line line, int offset, boolean inLocationDirection) {
+        if (inLocationDirection) {
+            return line.getGeoCoordinateAlongLine(offset);
+        } else {
+            return line.getGeoCoordinateAlongLine(line.getLineLength() - offset);
+        }
+
+    }
+
+    /**
+     * @param lrpLine                line on which lrp is placed
+     * @param lengthCoveredOnLrpLine distance along the line from the point on the line to which the LRP is projected to the relevant end based on bearing direction
+     * @param inLocationDirection    true: if bearing direction is same as the overall location direction
+     * @return Coordinates of the bearing destination
+     */
+    private GeoCoordinates getBearingDestination(Line lrpLine, int lengthCoveredOnLrpLine, boolean inLocationDirection, int bearingDistance) {
+        Line currentLine = lrpLine;
+        int lengthCovered = lengthCoveredOnLrpLine;
+
+        if (bearingDistance >= lengthCovered) {
+            List<Line> successors = calculateNextLinesInRoute(currentLine, inLocationDirection);
+            while (bearingDistance >= lengthCovered && isNotIntersection(successors.size())) {
+                currentLine = successors.get(0);
+                lengthCovered += currentLine.getLineLength();
+            }
```

Comment:
I'm not really convinced that this `while` loop will always terminate when we follow all the successors - I mean in case of a dead-end road.

Reply:
in case of dead-end node the isNotIntersection will fail because successor lines will be 0. but I can split it into multiple methods to make it more readable

Reply:
0975cd3318040870779bc3de4b706cd2ffb8b2ef